consensus/core/src/block.rs

```diff
@@ -253,6 +334,13 @@ impl VerifiedBlock {
         })
     }
 
+    /// Creates a new VerifiedBlock from a SignedBlock and the serialized bytes aren't available. Primarily this should be
+    /// used when proposing a new block and the bytes aren't available.
+    pub fn new_verified_unserialized(signed_block: SignedBlock) -> Result<Self, bcs::Error> {
+        let serialized = signed_block.serialize()?;
```

Comment:
Yeah that's basically I wanted to avoid with this method - I didn't want to give the option/alternative on someone accidentally passing another serialised version when proposing a new block . But I am ok removing if we don't see much of a risk.